### PR TITLE
s4747328 Improved 3D UNet

### PR DESCRIPTION
**Pull request by charlieeslack**
Name: Charles Slack
Student Number: 47473289

This is a pull request for COMP3710 Assignment 3, Task 4. This project is segmenting the Prostate 3D dataset using an improved 3D UNet, attempting to achieve an average DSC score of about 0.7 across all classes. This request currently acheives an average of 0.62 across classes.

### CODE REVIEW

**Comment by celinsze** [comment]
**This is an initial inspection, no action is required at this point**

Recognition problem: total 6.5 Marks
Problem Solving: (1)
•	Algorithm attempted for image segmentation and achieved an average Dice score of 0.61.
•	Unable to verify performance as individual Dice score, example output, and training metric plots are not provided. 
Algorithm implementation: (3)
•	Algorithm implemented in PyTorch for the UNet3D model and functions.
Design: Modularised and reusable code (1)
Commenting: Some comments throughout the code (0.5)
Difficulty: (1)
•	3D UNet – Normal 
•	Problem attempted but not solved

**Comment by gayanku** [comment]
## Marking
<table><tbody>
  <tr><td colspan='3'><b>Good/OK/Fair Practice (Design/Commenting, TF/Torch Usage)</b></td> </tr>
	<tr><td></td><td>Good design and implementation. </td><td></td></tr>
	<tr><td></td><td>Spacing and comments. </td><td></td></tr>
	<tr><td></td><td>No Header blocks. </td><td>-1</td></tr>
	
  <tr><td colspan='3'><b>Recognition Problem</b></td></tr>
	<tr><td></td><td>OK solution to problem. Mode performance not reported.</td><td>-2</td></tr>
	<tr><td></td><td>Driver Script present. </td><td></td></tr>
	<tr><td></td><td>File structure present. </td><td></td></tr>
	<tr><td></td><td>Good Usage & Demo & Visualisation & Data usage. </td><td></td></tr>
	<tr><td></td><td>Module present. </td><td></td></tr>
	<tr><td></td><td>Commenting missing. </td><td>-1</td></tr>
	<tr><td></td><td>No Data leakage found. </td><td></td></tr>
	<tr><td></td><td>Difficulty : Normal. UNet3D (Normal Difficulty)</td><td>-5</td></tr>
	
  <tr><td colspan='3'><b>Commit Log</b></td></tr>
	<tr><td></td><td>Some/Adequate Meaningful commit messages. </td><td>-1</td></tr>
	<tr><td></td><td>Some/Adequate Progressive commits. </td><td>-1</td></tr>
	
  <tr><td colspan='3'><b>Documentation</b></td></tr>
	<tr><td></td><td>Readme :Acceptable. No results, training graphs.</td><td>-2</td></tr>
	<tr><td></td><td>Model/technical explanation :Acceptable. </td><td>-2</td></tr>
	<tr><td></td><td>Description and Comments :Acceptable. No references.</td><td>-1</td></tr>
	<tr><td></td><td>Markdown used and PDF submitted. PDF checked.</td><td></td></tr>
	
  <tr><td colspan='3'><b>Pull Request</b></td></tr>
	<tr><td></td><td>Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch). On time submission.</td><td></td></tr>
	<tr><td></td><td>No Feedback required. </td><td></td></tr>
	<tr><td></td><td>Request Description is good. </td><td></td></tr>
  <tr><td colspan='2'><b>TOTAL</b></td><td>-16</td></tr>
</tbody></table>

Marked as per the due date and changes after which aren't necessarily allowed to contribute to grade for fairness.
_Subject to approval from Shakes_

**Comment by shakes76** [comment]
No visible indication of solving problem, short description and readme. -5
Will not be merged.
